### PR TITLE
DOC: refactor axes with lists

### PR DESCRIPTION
**Pull request by sciunto**
To avoid boilerplate code and to make things easier to modify.

### CODE REVIEW

File: doc/examples/segmentation/plot_marked_watershed.py [hunk 1]
```diff
@@ -46,21 +46,23 @@
...
+ax[1].set_title("Local Gradient")
+
+ax[2].imshow(markers, cmap=plt.cm.spectral, interpolation='nearest')
+ax[2].set_title("Markers")
+
+ax[3].imshow(image, cmap=plt.cm.gray, interpolation='nearest')
+ax[3].imshow(labels, cmap=plt.cm.spectral, interpolation='nearest', alpha=.7)
+ax[3].set_title("Segmented")
+
+for a in axes:
```
Comment (soupault) [inline doc/examples/segmentation/plot_marked_watershed.py]: *axes -> ax

**Comment by sciunto** [comment]
Fixed and I corrected few other places.

**Comment by sciunto** [comment]
and fixed again.

**Comment by soupault** [comment]
Here come the conflicts :smiling_imp:.

**Comment by sciunto** [comment]
@soupault rebased :8ball:

File: doc/examples/color_exposure/plot_ihc_color_separation.py [hunk 2]
```diff
@@ -61,9 +61,9 @@
 zdh = np.dstack((np.zeros_like(h), d, h))
 
 fig = plt.figure()
-ax = plt.subplot(1, 1, 1, sharex=ax0, sharey=ax0, adjustable='box-forced')
-ax.imshow(zdh)
-ax.set_title("Stain separated image (rescaled)")
-ax.axis('off')
+axis = plt.subplot(1, 1, 1, sharex=ax[0], sharey=ax[0], adjustable='box-forced')
```
Comment (soupault) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: We should refactor this in further PRs.

**Comment by soupault** [comment]
:+1:

File: doc/examples/color_exposure/plot_ihc_color_separation.py (hunk 2 above)
Comment (ahojnnes) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: I don't understand your comment?

Comment (ahojnnes) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: Ah, I see, this is a leftover comment. Ignore.

**Comment by ahojnnes** [comment]
@sciunto Travis fails.

File: doc/examples/color_exposure/plot_ihc_color_separation.py (hunk 2 above)
Comment (soupault) [inline doc/examples/color_exposure/plot_ihc_color_separation.py]: @ahojnnes I meant that `1, 1, 1, sharex=ax[0], sharey=ax[0]` is redundant here. `fig, axes` might be initialized in a single line (same as in other examples).

**Comment by sciunto** [comment]
@ahojnnes ? Travis is green for me.

**Comment by ahojnnes** [comment]
Interestingly it does for me now as wel. In it goes. Thanks.

**Comment by soupault** [comment]
Just to clarify, there was no magic - I restarted the build for OS X (the one which failed).

File: doc/examples/numpy_operations/plot_view_as_blocks.py [hunk 3]
```diff
@@ -45,26 +45,24 @@
...
-ax0, ax1, ax2, ax3 = axes.ravel()
+ax = axes.ravel()
 
-ax0.set_title("Original rescaled with\n spline interpolation (order=3)")
 l_resized = ndi.zoom(l, 2, order=3)
-
-ax0.imshow(l_resized, extent=(0, 128, 128, 0), interpolation='nearest',
+ax[0].set_title("Original rescaled with\n spline interpolation (order=3)")
+ax[0].imshow(l_resized, extent=(0, 128, 128, 0), interpolation='nearest',
            cmap=cm.Greys_r)
```
Comment (jni) [inline doc/examples/numpy_operations/plot_view_as_blocks.py]: This broke PEP8

Comment (ahojnnes) [inline doc/examples/numpy_operations/plot_view_as_blocks.py]: I'll quickly fix it.

Comment (ahojnnes) [inline doc/examples/numpy_operations/plot_view_as_blocks.py]: See https://github.com/scikit-image/scikit-image/pull/2136

**Comment by jni** [comment]
Can someone explain to me how this reduces boilerplate? The bulk of this PR is every `axn` being replaced with `ax[n]`. imho this *reduces* maintainability because you're still hardcoding a value, but now it's harder to grep/sed.

**Comment by ahojnnes** [comment]
Nope, but I don't care either way and thought this convention was agreed upon. I would argue that the majority of our examples use the `ax[N]` syntax, so rather than reducing boilerplate this increases consistency across the code base.

**Comment by jni** [comment]
@ahojnnes if that's the case I'm ok. I assumed that this PR changed *all* examples from one convention to the other. Consistency is good! =)

**Comment by ahojnnes** [comment]
@jni It's only 8 changed files and I hope we have more examples ;-)

**Pull request merged**